src/amplitude-client.js

```diff
@@ -1295,6 +1302,7 @@ AmplitudeClient.prototype._logEvent = function _logEvent(
       groups: utils.truncate(utils.validateGroups(groups)),
       group_properties: utils.truncate(utils.validateProperties(groupProperties)),
       user_agent: this._userAgent,
+      plan: plan,
```

Comment:
Do we need `_isObservePlanSet` if plan is optional? Can we instead directly use the current value?
```
plan: this.options.plan,
```

Reply:
yeah, if we have the default plan to be undefined, then we can reference from options.plan directly, although we still want to pass exactly what we need.